scripts/patch_websockets.py

```
@@ -0,0 +1,114 @@
+
+def _run_patch(ws_dir, patch_path, *, dry_run, reverse):
+    # -i reads the patch from a file so stdin is free; feeding stdin from
+    # /dev/null makes any GNU patch prompt default to "no" (skip), which is
+    # exactly the non-destructive behaviour we want for the dry-run probes.
+    cmd = ["patch", "-p1", "-i", patch_path]
+    if dry_run:
+        cmd.append("--dry-run")
+    if reverse:
+        cmd.append("--reverse")
+    with open(os.devnull, "rb") as devnull:
+        return subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)